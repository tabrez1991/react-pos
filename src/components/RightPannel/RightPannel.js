import React, { Fragment } from "react";
import { connect } from "react-redux";
import { addItem } from "../../actions/main"

const RightPannel = (props) => {
  return (
    <Fragment>
      <div className="product_list" data-test="product_list">
        {props.loader ?
          <div>Loading...</div>
          :
          props.pos_data.map((item, i) => (
            <div key={item.name} className={`product_product product_BG_${i + 1}`} onClick={() => props.addItem(item)} data-test="product_product">
              <img src={require("../../images/" + item.image)} />
              <h5>{item.name}</h5>
              <div className="hovered-details">
                <div>{item.price} EUR</div>
                <div>{item.description}</div>
              </div>
            </div>
          ))
        }
      </div>
    </Fragment>
  );
};

const mapStateToProps = (state) => ({
  loader: state.main.loader,
  pos_data: state.main.pos_data
});

export default connect(mapStateToProps, { addItem })(RightPannel);
