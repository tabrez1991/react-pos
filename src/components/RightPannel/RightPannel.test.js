import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAtrr, testStore } from '../../utils/Tests';
import RightPannel from "./RightPannel";

const setUp = (initialState = {}) => {
    const store = testStore(initialState);
    const wrapper = shallow(<RightPannel store={store} />).childAt(0).dive();
    return wrapper;
};

describe('RightPannel Component', () => {

    let wrapper;
    beforeEach(() => {
        const initialState = {
            loader:true,
            pos_data:[{"name": "comuter", "price": "130", "description": "","image": "comuter.jpg"},{"name": "comuter", "price": "130", "description": "","image": "comuter.jpg"}]
        }
        wrapper = setUp(initialState);
    });

    it('Should render without errors', () => {
        const component = findByTestAtrr(wrapper, 'product_list');
        expect(component.length).toBe(1);
    });

     it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});