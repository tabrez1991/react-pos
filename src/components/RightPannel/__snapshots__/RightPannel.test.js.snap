// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightPannel Component should match the snapshot 1`] = `"<div class=\\"product_list\\" data-test=\\"product_list\\"><div>Loading...</div></div>"`;
